src/rascal/structure_managers/property_block_sparse.hh

```diff
@@ -580,21 +580,19 @@ namespace rascal {
     //! Property accessor by cluster ref
     template <size_t CallerLayer>
     InputData_t & operator[](const ClusterRefKey<Order, CallerLayer> & id) {
-      static_assert(CallerLayer >= PropertyLayer,
-                    "You are trying to access a property that does not exist at"
-                    "this depth in the adaptor stack.");
-
-      return this->operator[](id.get_cluster_index(CallerLayer));
+      // You are trying to access a property that does not exist at this depth
+      // in the adaptor stack.
+      assert(static_cast<int>(CallerLayer) >= this->get_property_layer());
```

Comment:
Can these stay `static_assert`? By making `get_property_layer` constexpr for example. 

It would make for a better dev experience to get the error at compile-time instead of runtime, and only in debug mode.

Reply:
No, sorry, the `Layer` is not a template argument any more, hence we cannot check at compiletime.

From `property_base.hh`:
```
//! constructor
    PropertyBase(StructureManagerBase & manager, Dim_t nb_row, Dim_t nb_col,
                 size_t order, size_t layer,
                 std::string metadata = "no metadata")
        : base_manager{manager}, nb_col{nb_col}, nb_row{nb_row},
          nb_comp{nb_row * nb_col}, order{order},
          property_layer{layer}, metadata{metadata} {}
```
and therefore
```
//! returns the property layer
    Dim_t get_property_layer() const { return this->property_layer; }
```

Reply:
I'm dumb, I'll take it all back :p

Will of course change it correspondingly

Reply:
Ok, should be fine then :smiley: